### DISCUSSION TITLE
CMake CGAL Example for a simple project

### DISCUSSION THREAD

**Discussion by petrasvestartas** [CGAL/cgal#6657] [General]
Hi,

I am currently learning CMake by watching a lot of Udemy tutorials.
It is good to go through the basics and more complex linking examples.
Nevertheless, I still feel that I am relatively new to CMake language specially when I want to link libraries such as CGAL, Eigen or Boost.

Can someone show a basic example how do you typically link CGAL through CMake?
There is a bunch of CMakeLists.txt and cmake files because CGAL is huge.
I am especially curious, how do normally install new releases? Do you always manually go to release section of github and download the files or there is an automated procedure.

**Comment by lrineau** [comment]
> Can someone show a basic example how do you typically link CGAL through CMake?

There is an example at https://doc.cgal.org/latest/Manual/devman_create_and_use_a_cmakelist.html.


> I am especially curious, how do normally install new releases? Do you always manually go to release section of github and download the files or there is an automated procedure.

That depends on your operating system. There are package managers that can install CGAL. See the documentation: https://doc.cgal.org/latest/Manual/general_intro.html

Which operating system do you use?

**Reply by petrasvestartas** [reply]
* I am using windows.  Also I am stuck with Mac. But for now I would like to learn how to create C++ exe that references CGAL using cmake for Windows which is shown in your example.

* Is there an automatic procedure to fetch / download latest CGAL package through CMAKE or I must do : git clone -> cmake -- install . -> Configure my CMakeLists according to your example?

**Reply by lrineau** [reply]
The CMake API is as simple as:
```cmake
find_package(CGAL)
add_executable(my_executable my_source_file.cpp)
target_link_libraries(my_executable CGAL::CGAL)
```

> * Is there an automatic procedure to fetch / download latest CGAL package through CMAKE or I must do : git clone -> cmake -- install . -> Configure my CMakeLists according to your example?

Once you have a clone of CGAL, then set the environment or CMake variable `CGAL_DIR` to the root of your clone. And that is it. When you want to update CGAL, just `git pull` inside the cloned repository.

**Reply by petrasvestartas** [reply]
Thank you. This helps a lot.

Could you show how you define in your computer CGAL_DIR once you configure a new exe project with CGAL in the CMakeLists.txt?

**Reply by lrineau** [reply]
- If you use CMake GUI, you can add a variable that way: https://cmake.org/cmake/help/latest/guide/user-interaction/index.html#id12
- If you want to add a permanent `CGAL_DIR` environment variable; https://www.tenforums.com/tutorials/121664-set-new-user-system-environment-variables-windows.html
